traced/tests/src/stopgap/guard_fail.rs

```diff
@@ -222,8 +222,43 @@ fn trace_looping_nested() {
 }
 
 #[bench]
-fn bench_looping_nested(b: &mut Bencher) {
+fn bench_looping(b: &mut Bencher) {
+    struct InterpCtx {
+        compiled: bool,
+        count: u8
+    }
+
+    #[interp_step]
+    fn interp_step(io: &mut InterpCtx) -> bool {
+        if io.compiled {
```

Comment:
The reason why this benchmark doesn't work, is that the stopgrap interpreter is never run. You compiled `interp_step` with `io.compiled = false`, so it compiles the `else` branch on this function. Then you set `io.compiled = true` which I assume you did to immediately invoke the stopgap interpreter to benchmark it. However, `ct.execute` doesn't automatically run the stopgap interpreter, rather it returns a pointer to the stopgap interpreter which we can then cast to a `StopgapInterpreter` and run it (see https://github.com/softdevteam/yk/blob/master/traced/ykrt/src/mt.rs#L281 how this is done).

Keep in mind that the stopgap interpreter only finishes a single instruction, in this case `if io.compiled`, and then stops, so even if you run the stopgap interpreter, the `assert` below will never pass.

Reply:
Dumb question: which tests in `tests/src/stopgap` do execute the stopgap interpreter?

Reply:
`recursion1` and `recursion2` test that the stopgap interpreter executes. The other tests check the initialisation of the stopgap interpreter (which is kinda hidden in `ct.execute`). There's also `tests/src/stopgap/interp.rs` which have more implementation specific tests of the stopgap interpreter. And then of course there's the tests in `ykrt/src/mt.rs` which test the whole meta-tracing chain, i.e. interpret->compile->guardfail->stopgap->interpret.